syntax = "proto3";

import "google/protobuf/empty.proto";

service PepperAuto {
    rpc GetImg(google.protobuf.Empty) returns (ImageChunk);

    rpc ExecuteAction(ExecuteParam) returns (ConfirmationChunk);

    rpc RobotSay(SentenceParam) returns (ConfirmationChunk);
}

message ImageChunk {
    bytes image_data=1;
}

message ConfirmationChunk {
    bool confirmed=1;
}

// Joint name: HeadYaw, HeadPitch, LShoulderPitch, LShoulderRoll, LElbowYaw, LElbowRoll, LWristYaw, LHand, RShoulderPitch, RShoulderRoll, RElbowYaw, RElbowRoll, RWristYaw, RHand, HipRoll, HipPitch, KneePitch, AnklePitch
// angle in degrees
// speed between 0 and 1
message ExecuteParam {
  string joint_name=1;
  float angle=2;
  float speed=3;
}

message SentenceParam {
    string sentence_to_say=1;
}
